hearth/lib/hearth/cbor/half.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Hearth
+  module Cbor
+    # 16 bit IEEE 754 half-precision floats
+    # Support decoding only
+    # format:
+    # sign - 1 bit
+    # exponent - 5 bits
+    # precision - 10 bits
+    module Half
```

Comment:
I really think this module is not necessary and can just be in the half method in the decoder.

Reply:
Part of my orig thought was to try and reduce the overall complexity of the decoder class and isolate logic a bit.  But I think the decoder is WAY past that point anyway.